frontend/src/components/common/Resource/Resource.tsx

```diff
@@ -894,3 +895,190 @@ export function ConditionsSection(props: { resource: KubeObjectInterface | null
     </SectionBox>
   );
 }
+
+export interface VolumeSectionProps {
+  volume?: KubeVolume;
+  resource: KubeObjectInterface | null;
+}
+
+export interface PrintVolumeLinkProps {
+  volumeName: string;
+  volumeKind: string;
+  volume: KubeVolume;
+}
+
+export interface VolumePrints {
+  directPrint?: any[];
+  yamlPrint?: any[];
+}
+
+export function VolumeSection(props: VolumeSectionProps) {
+  const { t } = useTranslation();
+  const { resource } = props;
+
+  const volumes = resource?.spec?.volumes;
+
+  if (!volumes) {
+    return null;
+  }
+
+  const namespace = resource?.metadata?.namespace;
+
+  /*
+   * printVolumeLink will print a working link that is set within the router using fields from the resource as params
+   */
+  function PrintVolumeLink(props: PrintVolumeLinkProps) {
+    const { volumeName, volumeKind, volume } = props;
+    const resourceClasses = ResourceClasses;
+    const classList = Object.keys(ResourceClasses);
+
+    for (const kind of classList) {
+      if (kind.toLowerCase() === volumeKind.toLowerCase()) {
+        const VolumeClass = resourceClasses[kind];
+        const volumeInstance = new VolumeClass(volume);
+        const volumeRoute = volumeInstance.detailsRoute;
+        const volumeNamespace = volumeInstance.isNamespaced ? namespace : null;
+        // we use the keys of the volume[volumeKind], which is the non class created volume item, to find the name of the volume source
+        const volumeKeys = Object.keys(volume[volumeKind]);
+        let detailName;
+
+        // where the volume[volumeKind] contains a "name" word in the keys, grab that value
+        for (const key of volumeKeys) {
+          if (key.includes('name')) {
```

Comment:
This is a very implicit and we cannot guarantee that the name you find this way is the right name of the volume kind.
So we should make this explicit: maybe having an object corresponding a volume kind with the name key, like
```typescript
const volumeKindNames = {
  configMap: 'name',
  secret: 'secretName',
};
```

This way we know exactly what is supposed to work and what is not.